src/GameMachine.java

```diff
@@ -9,27 +10,21 @@
             new Game("Starcraft", 35)
     );
 
-    public List<Game> getGames() {
-        return games;
-    }
-
-    public void setGames(List<Game> games) {
-        this.games = games;
-    }
+    void buyGame(String gameName, double payment) throws NoSuchGameException, NotEnoughtMoneyException {
+        Optional<Game> foundGame = games.stream()
+                .filter(game -> game.getName().equals(gameName))
+                .findFirst();
+        if (foundGame.isEmpty()) {
```

Comment:
Instead of this if you could finish your stream like this -> .findFirst().orElseThrow(() -> new NoSuchGameException("no game)); then your if is not needed and code is much cleaner :)

Reply:
I made the change but i had to replace optional by Game object.